pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -389,7 +390,7 @@ buildStdenv.mkDerivation ({
     (enableFeature enableDebugSymbols "debug-symbols")
   ]
   ++ lib.optionals enableDebugSymbols [ "--disable-strip" "--disable-install-strip" ]
-  ++ lib.optional enableOfficialBranding "--enable-official-branding"
+  ++ lib.optionals enableOfficialBranding [ "--enable-official-branding" "--with-branding=browser/branding/${branding}" ]
```

Comment:
Ideally we could stick with the default branding value for most consumers of `buildMozillaMach`.

Please note that this function serves firefox, firefox-esr, librewolf, thunderbird, soon apparently also betterbird, and at least the mail clients are left out here. 

If we make the branding overridable, then the whole path (`browser/branding/official`) needs to be configurable, not just its final segment.

For most causes not setting the branding does work, so I would only want it to be set in cases where it does not work out.